Update release note with OpenAPI changes (#2174)
ballerina-platform/ballerina-release: release-notes/swan-lake-release-notes/swan-lake-2201.2.3-release-note.md

* Add a new quick-fix code action to fill in the missing required fields of a mapping constructor
* Add a new module-level completion item to generate an expression-bodied function snippet

## Improvements

### OpenAPI Tool

> - Add improvement to OpenAPI client configuration handling. The record which contains the information for client initialization has changed from ClientConfig to ConnectionConfig. Few of defaultable configuration fields which were in ClientConfig have been changed to the optional fields in ConnectionConfig record. Due to this change ConnectionConfig record can be defined as a configurable variable.

Reviewer: ```suggestion
-Improved the OpenAPI client configuration handling functionality. Now, the record, which contains the information for client initialization is changed from the `ClientConfig` to `ConnectionConfig`. A few of the defaultable configuration fields, which were in the `ClientConfig` have been changed to the optional fields in the `ConnectionConfig` record. Due to this change, the `ConnectionConfig` record can be defined as a configurable variable.
```
